src/tests/class_tests/smartpeak/source/Utilities_test.cpp

```diff
@@ -382,52 +382,48 @@ BOOST_AUTO_TEST_CASE(endsWith)
-  BOOST_CHECK_EQUAL(c[1][43], "22"); // number of items within the folder
-  BOOST_CHECK_EQUAL(c[2][43], "Directory");
+  std::filesystem::path pathname = SMARTPEAK_GET_TEST_DATA_PATH("");
+  
+  std::array<std::vector<std::string>, 4> directory_entries;
+  
+  std::tuple<std::string, std::string> order_by_name = std::make_tuple ("name", "ascending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_name);
+  BOOST_CHECK_EQUAL(directory_entries[0][0], "170808_Jonathan_yeast_Sacc1_1x_1_FluxTest_1900-01-01_000000.featureXML");
+  directory_entries[0].clear(); directory_entries[1].clear();
+  directory_entries[2].clear(); directory_entries[3].clear();
+  
+  std::tuple<std::string, std::string> order_by_file_extension = std::make_tuple ("extension", "descending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_file_extension);
+  BOOST_CHECK_EQUAL(directory_entries[2][0], "Directory");
+  BOOST_CHECK_EQUAL(directory_entries[2][directory_entries.size()-1], ".txt");
+  directory_entries[0].clear(); directory_entries[1].clear();
+  directory_entries[2].clear(); directory_entries[3].clear();
+  
+  std::tuple<std::string, std::string> order_by_size = std::make_tuple ("size", "descending");
+  directory_entries = Utilities::getFolderContents(pathname, order_by_size);
+  BOOST_CHECK_EQUAL(directory_entries[1][0], "3804256");
+  BOOST_CHECK_EQUAL(directory_entries[1][directory_entries[1].size()-3], "192");
+  directory_entries[0].clear(); directory_entries[1].clear();
+  directory_entries[2].clear(); directory_entries[3].clear();
 }
 
-BOOST_AUTO_TEST_CASE(getParentPathname)
-{
-#ifdef _WIN32
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("D://///"), "D:/");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("D:\\"), "D:");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("D:/"), "D:");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("D:"), "");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("E:/home/user/docs"), "E:/home/user");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("E://home///user//docs"), "E://home///user");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("E:/home/user/docs and a space"), "E:/home/user");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("E:/home/user/docs/"), "E:/home/user/docs");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("//home///user//docs"), "//home///user");
-#else
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname(""), "");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/"), "");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/home/user/docs and a space"), "/home/user");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/home/user/docs/"), "/home/user/docs");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/home/user/docs"), "/home/user");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/home"), "/");
-  BOOST_CHECK_EQUAL(Utilities::getParentPathname("/home/file.txt"), "/home");
-#endif
+
+BOOST_AUTO_TEST_CASE(getParentPath)
+{  
```

Comment:
If you can include the Windows like paths similar to the tests for the previous version of this function, that would be great.

Reply:
I have updated the test to include similar test as in the previous version.